src/schema/posts.ts

```diff
@@ -508,6 +523,28 @@ export const typeDefs = /* GraphQL */ `
       comment: String
     ): EmptyResponse @auth
 
+    """
+    To allow user to edit posts
+    """
+    editPost(
+      """
+      ID of the post to update
+      """
+      id: ID!
+      """
+      Avatar image for the squad
+      """
+      image: Upload
+      """
+      Title of the post (max 80 chars)
+      """
+      title: String
+      """
+      Content of the post (max 4000 chars)
+      """
+      content: String
+    ): EmptyResponse! @auth
```

Comment:
Just wanted to lessen the query since the updates considered in this mutation will be based on all the inputs which the consuming end already has. With that, fetching one post shouldn't hurt the performance, let's return it as you suggested.

Reply:
This is a valid way of thinking and I appreciate it. Bare in mind that the markdown and mentions are not provided as inputs so it will be missing